components/registration-framework/org.wso2.carbon.identity.user.registration.mgt/src/main/java/org/wso2/carbon/identity/user/registration/mgt/dao/RegistrationFlowDAOImpl.java

```diff
@@ -242,8 +232,220 @@ public RegistrationFlowDTO getDefaultRegistrationFlowByTenant(int tenantId) thro
+                    });
+
+            // If no rows found, return null
+            if (rows == null || rows.isEmpty()) {
+                return null;
+            }
+
+            // Aggregate rows into RegistrationGraphConfig
+            return mapRowsToGraphConfig(rows, tenantId);
+        } catch (DataAccessException e) {
+            throw handleServerException(Constants.ErrorMessages.ERROR_CODE_GET_REG_GRAPH_FAILED, e, tenantId);
+        }
+    }
+
+    private RegistrationGraphConfig mapRowsToGraphConfig(List<Map<String, Object>> rows, int tenantId)
+            throws RegistrationServerException {
+
+        RegistrationGraphConfig registrationGraphConfig = new RegistrationGraphConfig();
+        Map<String, NodeConfig> nodeConfigs = new HashMap<>();
+        Map<String, StepDTO> nodePageMappings = new HashMap<>();
+
+        for (Map<String, Object> row : rows) {
+            // Set flow ID once.
+            if (registrationGraphConfig.getId() == null) {
+                registrationGraphConfig.setId((String) row.get(DB_SCHEMA_ALIAS_FLOW_ID));
+            }
+
+            // Process node configuration.
+            String nodeUuid = (String) row.get(DB_SCHEMA_COLUMN_NAME_NODE_ID);
+            if (nodeUuid != null) {
+                nodeConfigs.putIfAbsent(nodeUuid, new NodeConfig.Builder()
+                        .id(nodeUuid)
+                        .type((String) row.get(DB_SCHEMA_COLUMN_NAME_NODE_TYPE))
+                        .isFirstNode((Boolean) row.get(DB_SCHEMA_COLUMN_NAME_IS_FIRST_NODE))
+                        .build());
+
+                NodeConfig nodeConfig = nodeConfigs.get(nodeUuid);
+
+                // Update first node of the graph.
+                if (nodeConfig.isFirstNode() && registrationGraphConfig.getFirstNodeId() == null) {
+                    registrationGraphConfig.setFirstNodeId(nodeUuid);
+                }
+
+                // Process node edges.
+                String nextNodeUuid = (String) row.get(DB_SCHEMA_ALIAS_NEXT_NODE_ID);
+                if (nextNodeUuid != null) {
+                    nodeConfig.getEdges().add(new NodeEdge(
+                            nodeUuid,
+                            nextNodeUuid,
+                            (String) row.get(DB_SCHEMA_COLUMN_NAME_TRIGGERING_ELEMENT)
+                    ));
+                }
+
+                // Process executor details.
+                String executorName = (String) row.get(DB_SCHEMA_COLUMN_NAME_EXECUTOR_NAME);
+                if (executorName != null) {
+                    nodeConfig.setExecutorConfig(new ExecutorDTO(
+                            executorName,
+                            (String) row.get(DB_SCHEMA_COLUMN_NAME_IDP_NAME)
+                    ));
+                }
+            }
+
+            // Process page content.
+            String stepId = (String) row.get(DB_SCHEMA_COLUMN_NAME_STEP_ID);
+            if (stepId != null) {
```

Comment:
what does step id being null mean in this context? Does that warrant a server error?